Create website layout

ibaniya98 opened the pull request:
- Created a landing page
 
![image](https://user-images.githubusercontent.com/43834826/106044424-b52a7780-60ad-11eb-97c6-286a0641713f.png)

- Added responsive navbar to application

| Desktop Version | Mobile Version |
|-----------------|----------------|
|     <img src="https://user-images.githubusercontent.com/43834826/106005305-189eb000-6082-11eb-89b0-d0f933de3637.png" alt="drawing" width="300"/>            | <img src="https://user-images.githubusercontent.com/43834826/106005353-25bb9f00-6082-11eb-85d1-3df34cb99671.png" alt="drawing" width="200"/> |

- Create a `PrivateLayout` component that can be re-used to create any page a user sees after logging in.
- Added utils method to get the screen size since Next.js does SSR and does not have access to the `window` object before rendering it on the client.

ibaniya98 committed:
Create navbar for desktop view

ibaniya98 committed:
Create layout for desktop view

ibaniya98 committed:
Add navbar for mobile

ibaniya98 committed:
Adjust layout width for page contents

ibaniya98 committed:
Use fixed navbar for desktop version

ibaniya98 committed:
Create landing page

vrajp13 reviewed (changes requested):
The desktop navbar is great, the issue is that the home page and calendar page can be accessed, however, the analytics page and user profile is showing up as 404 error page. So not sure if you want to create a layout for that as well. Also, how do I check the mobile layout?

ibaniya98 committed:
Remove unused navigation items

ibaniya98 commented:
I just added those to see how our final application looks. Removed them for now since we do not have those pages yet and can add them later in the sprint.
There are two ways to view the mobile view
1. Use chrome devtools to view in different devices (https://developers.google.com/web/tools/chrome-devtools/device-mode)
2. The nice thing about integrating with Vercel is it creates a preview of the PR changes. For this PR, the URL is https://tiptracker-git-frontend-add-website-layout.nepal-6d48a0.vercel.app/ You can open that link from any device you want

vrajp13 reviewed (approved):
Great, both the desktop and mobile layouts are working correctly.
